kafka/source/pkg/reconciler/source/kafkasource.go

```diff
@@ -112,6 +113,8 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, src *v1beta1.KafkaSource
 	sinkURI, err := r.sinkResolver.URIFromDestinationV1(*dest, src)
 	if err != nil {
 		src.Status.MarkNoSink("NotFound", "")
+		//delete adapter deployment if sink not found
+		r.deleteReceiveAdapter(ctx, src)
```

Comment:
Related to comment below, the error should be returned in the function and it should be handled here

Reply:
have handled the error in the deleteReceiveAdapter func, please let me know if thats ok ?